AppLovinAdapter/src/main/java/com/chartboost/mediation/applovinadapter/AppLovinAdapter.kt

```diff
@@ -668,6 +676,29 @@ class AppLovinAdapter : PartnerAdapter {
                         fullyWatched: Boolean
                     ) {
                         // TODO: HB-4119: Need to check if we need to add logic here for rewards.
+                        // Only stop the reward if the user was explicity deny listed by AppLovin
+                        if (isUserVerified == false) {
+                            PartnerLogController.log(
+                                CUSTOM,
+                                "Unable to reward due to user being denylisted by AppLovin."
+                            )
+                            return
+                        }
+
+                        if (!fullyWatched) {
+                            PartnerLogController.log(
+                                CUSTOM,
+                                "Unable to reward due to video not being fully watched."
+                            )
+                            return
+                        }
```

Comment:
Nit: Could we use a `when` here for readability? 

Also, another idea: `require(isUserVerified)` (same for `fullyWatched`) instead of `if` + inverted condition + `return` might make it easier to understand immediately.

Reply:
I'll use the `when`. We don't know how AppLovin's SDK will handle that callback throwing an exception